### PR TITLE
plumbing for setStatusLight

### PR DESCRIPTION
**Pull request by tballmsft**

### CODE REVIEW

**Comment by pelikhan** [comment]
You should be able to set the setStatusLight using a writeRegister on any device.

**Comment by tballmsft** [comment]
Then what is the purpose of this statement?

**Comment by pelikhan** [comment]
It's a command that's common to all services so i factored it out.

**Comment by pelikhan** [comment]
[cid:38d49c42-3abb-4163-81f8-fc4c55d2bec7]


Sent from Outlook<http://aka.ms/weboutlook>

________________________________

File: src/vm/environment.ts [hunk 1]
```diff
@@ -183,6 +183,23 @@ export class VMEnvironment
...
     }
 
+    public async setStatusAsync(color: number) {
+        for(const role of Object.keys(this._envs)) {
+            const serv = this._envs[role]
+            if (serv) {
+                const device = serv.service?.device
+                if (device) {
+                    const ledController = device.statusLight
+                    // const vals = inputs.map(i => parseInt(i.value))
```
Comment (pelikhan) [inline src/vm/environment.ts:193]: use blink command on LED controller

File: src/vm/runner.ts [hunk 2]
```diff
@@ -52,6 +52,7 @@ export interface VMEnvironmentInterface {
         values: atomic[]
     ) => Promise<void>
     lookupAsync: (e: jsep.MemberExpression | string) => Promise<atomic>
+    setStatusAsync: (status: number) => Promise<void>
```
Comment (pelikhan) [inline src/vm/runner.ts:55]: should take a role argument as well

File: src/vm/runner.ts [hunk 3]
```diff
@@ -196,6 +197,12 @@ class VMCommandEvaluator {
                     ? VMInternalStatus.Completed
                     : VMInternalStatus.Running
             }
+            case "setStatus": {
+                // send command to all devices
```
Comment (pelikhan) [inline src/vm/runner.ts:201]: the API takes a role argument

**Pull request closed without merging**